src/LuaPrinter.ts

```diff
@@ -447,14 +479,34 @@ export class LuaPrinter {
 
         const chunks: SourceChunk[] = [];
 
-        chunks.push("function(");
+        chunks.push("function");
+
+        if (name) {
+            chunks.push(" ");
+            chunks.push(name);
+        }
+
+        chunks.push("(");
         chunks.push(...this.joinChunks(", ", parameterChunks));
-        chunks.push(")\n");
+        chunks.push(")");
+
+        if (expression.body.statements
+            && expression.body.statements.length === 1
+            && tstl.isReturnStatement(expression.body.statements[0])
+            && (expression.flags & tstl.FunctionExpressionFlags.Inline) !== 0)
+        {
+            // Inline return-only functions with the flag
+            chunks.push(" ");
+            chunks.push(this.printReturnStatement(expression.body.statements[0] as tstl.ReturnStatement, true));
```

Comment:
Is this `as tstl.ReturnStatement` required? Or does the typeguard in the `if` condition not work on index expressions?

Reply:
Yeah, it's required - the typeguard doesn't work with array elements